handler/src/main/java/io/netty/handler/ssl/ReleasableOpenSslEngine.java

```diff
@@ -0,0 +1,1995 @@
+                    break;
+                case REQUIRE:
+                    SSL.setVerify(ssl, SSL.SSL_CVERIFY_REQUIRE, OpenSslContext.VERIFY_DEPTH);
+                    break;
+                case OPTIONAL:
+                    SSL.setVerify(ssl, SSL.SSL_CVERIFY_OPTIONAL, OpenSslContext.VERIFY_DEPTH);
+                    break;
+            }
+            clientAuth = mode;
+        }
+    }
+
+    @Override
+    public final void setEnableSessionCreation(boolean b) {
+        if (b) {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    @Override
+    public final boolean getEnableSessionCreation() {
+        return false;
+    }
+
+    @Override
+    public final synchronized SSLParameters getSSLParameters() {
+        SSLParameters sslParameters = super.getSSLParameters();
+
+        int version = PlatformDependent.javaVersion();
+        if (version >= 7) {
+            sslParameters.setEndpointIdentificationAlgorithm(endPointIdentificationAlgorithm);
+            SslParametersUtils.setAlgorithmConstraints(sslParameters, algorithmConstraints);
+            if (version >= 8) {
+                if (SET_SERVER_NAMES_METHOD != null && sniHostNames != null) {
+                    try {
+                        SET_SERVER_NAMES_METHOD.invoke(sslParameters, sniHostNames);
+                    } catch (IllegalAccessException e) {
+                        throw new Error(e);
+                    } catch (InvocationTargetException e) {
+                        throw new Error(e);
+                    }
+                }
+                if (SET_USE_CIPHER_SUITES_ORDER_METHOD != null && !isDestroyed()) {
+                    try {
+                        SET_USE_CIPHER_SUITES_ORDER_METHOD.invoke(sslParameters,
+                                (SSL.getOptions(ssl) & SSL.SSL_OP_CIPHER_SERVER_PREFERENCE) != 0);
+                    } catch (IllegalAccessException e) {
+                        throw new Error(e);
+                    } catch (InvocationTargetException e) {
+                        throw new Error(e);
+                    }
+                }
+            }
+        }
+        return sslParameters;
+    }
+
+    @Override
+    public final synchronized void setSSLParameters(SSLParameters sslParameters) {
+        super.setSSLParameters(sslParameters);
+
+        int version = PlatformDependent.javaVersion();
+        if (version >= 7) {
+            endPointIdentificationAlgorithm = sslParameters.getEndpointIdentificationAlgorithm();
+            algorithmConstraints = sslParameters.getAlgorithmConstraints();
+            if (version >= 8) {
+                if (SNI_HOSTNAME_CLASS != null && clientMode && !isDestroyed()) {
+                    assert GET_SERVER_NAMES_METHOD != null;
+                    assert GET_ASCII_NAME_METHOD != null;
+                    try {
+                        List<?> servernames = (List<?>) GE
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS134)